pkg/scheduler/worker/workerpool_test.go

```diff
@@ -87,3 +84,77 @@ func TestWorkerPool(t *testing.T) {
 	require.Equal(t, reconEntity.SchedulingID, testInvoker.params[0].SchedulingID)
 	require.Equal(t, opsProcessable[0].CorrelationID, testInvoker.params[0].CorrelationID)
 }
+
+func TestWorkerPoolParallel(t *testing.T) {
+
+	t.Run("Multiple WorkerPools watching same reconciliation repository", func(t *testing.T) {
+
+		//initialize WaitGroup
+		var wg sync.WaitGroup
+		//prepare keb clusters
+		kebClusters := []*keb.Cluster{keb.NewCluster(t, strconv.Itoa(1), 1, false, keb.OneComponentDummy),
+			keb.NewCluster(t, strconv.Itoa(2), 1, false, keb.OneComponentDummy),
+			keb.NewCluster(t, strconv.Itoa(3), 1, false, keb.OneComponentDummy),
+		}
+
+		//create mock database connection
+		testDB := db.NewTestConnection(t)
+
+		//create cluster inventory
+		inventory, err := cluster.NewInventory(testDB, true, &cluster.MetricsCollectorMock{})
+		require.NoError(t, err)
+
+		//add clusters to inventory
+		var clusterStates [3]*cluster.State
+		for i := range kebClusters {
+			clusterStates[i], err = inventory.CreateOrUpdate(1, kebClusters[i])
+		}
+		require.NoError(t, err)
+
+		//create test invoker to be able to verify invoker calls and update operation state
+		testInvoker := &testInvoker{errChannel: make(chan error, 100)}
+
+		//create reconciliation for cluster
+		testInvoker.reconRepo, err = reconciliation.NewPersistedReconciliationRepository(testDB, true)
+		require.NoError(t, err)
+		for i := range clusterStates {
+			_, err = testInvoker.reconRepo.CreateReconciliation(clusterStates[i], nil)
+		}
+		require.NoError(t, err)
+		opsProcessable, err := testInvoker.reconRepo.GetProcessableOperations(0)
+		require.Len(t, opsProcessable, 3) // only first prio
+		require.NoError(t, err)
+
+		//initialize worker pool
+		workerPool, err := NewWorkerPool(&InventoryRetriever{inventory}, testInvoker.reconRepo, testInvoker, nil, logger.NewLogger(true))
+		require.NoError(t, err)
+
+		//create time limited context
+		ctx, cancelFct := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancelFct()
+
+		//start multiple worker pools
+		errChannel := make(chan error)
+		startAt := time.Now().Add(1 * time.Second)
+		for i := 0; i < 3; i++ {
+			wg.Add(1)
+			go func(errChannel chan error, workerPool *Pool) {
+				defer wg.Done()
+				time.Sleep(time.Until(startAt))
+				err := workerPool.Run(ctx)
+				if err != nil {
+					errChannel <- err
+				}
+			}(errChannel, workerPool)
+		}
+		wg.Wait()
+
+		//verify that invoker was properly called
+		require.Equal(t, 12, len(testInvoker.errChannel))
```

Comment:
Would be good to add a comment why we check for 12 to make it more obvious for colleagues who don't know all details of the execution flow